### ISSUE TITLE
How to use?

### ISSUE THREAD

**Issue by bdillahu** [yoursamlan/pubsidian#1]
Sorry, not really a *.js person... can you give a pointer how to get started?

**Comment by yoursamlan** [comment]
Hi!
This project is currently under development. I'm trying to create a one-click deployment as soon as possible.

Meanwhile, if you want to host your own notes using pubsidian, you have to follow the steps below.
1. First of all you have to convert your obsidian notes from .md to .html format. You can either manually do that, or you can perform the following steps.
    0.  **Make sure, you have installed Python 3.5 or later.**
    a.  Now, **fork and download** [this repository](https://github.com/yoursamlan/pubsidian) to your local machine.
    b. After downloading the repository, go to the ```tools``` directory.
    c. Now open command prompt/ terminal and run
        ``` pip install requirements.txt ```
    d. After installing the libraries successfully, copy ```md2html.py``` from the tools directory to the obsidian directory, [i.e where your notes (.md) are stored.]
    e. Now open the command prompt again and run
    ``` python md2html.py```
    
    After running successfully, you'll see something like this...
    
![image](https://user-images.githubusercontent.com/33586885/128258717-dfb04e85-2ea0-478c-bbc9-8f002865f8b4.png)

- Your .md notes now have been converted to .html successfully. These .html notes will be stored in the ```pages``` directory. Also, you will see another ```data.json``` in your main directory.
    
2. Now copy and replace your generated ```data.json``` and ```pages``` directory in the forked directory and push it to the github.
3. Now enable github-pages from the setting.

And Voila, your notes are online. :)


---
If you have any question regarding any step, feel free to ask. If not, I will close this thread after 24 hr.

**Comment by arnouthpessoa** [comment]
Hey master, all good?

Im not a js person, neither coder. Im an UX Designer, and I'm trying to publish my obsidian vault without success because every method need to know how to code, even a little.
I tried your one-click method but failed like https://github.com/yoursamlan/pubsidian/issues/11: stuck in 1~2 notes and nothing happens anymore.
So I'm trying to follow your steps in this guide, but have failed in:

pip install requirements.txt

I open the cmd in the tools folder, inside pubsidian-main, but pip and install command is not recognized. What I'm doing wrong?

Thank you in advance!

**Comment by yoursamlan** [comment]
Greetings @arnouthpessoa ,

I'm sorry you're facing problems. But to understand the scenario better, I need few more information, like... 

Which installation are you using ? I'm asking because the method, mentioned above is deprecated. I had built a GUI for specifically for windows. You can download and install it from [here](https://github.com/yoursamlan/pubsidian/releases/download/Windows/PubsidianConverter1.0alpha.exe). Detailed instructions have been mentioned in the ```Quickstart Guide``` section of the ```readme.MD```. Try it and do let me know, if you face any issue.

Thank you and have a great day :)

**Comment by arnouthpessoa** [comment]
![image](https://user-images.githubusercontent.com/113953046/191310291-12c20743-a153-4442-9e5a-bc8973c74a08.png)

Hey Mr @yoursamlan!

This is what happens, unfortunately! The script can only convert 2 files and its done. Previously, the script stopped in converting one file and nothing more. Now, it finished his job but without convert the whole folder. =/

Thank you again! I'm trying to convert my vault to do presentations of my work without using the Notion(Obsidian Rulez!!) xD

**Comment by yoursamlan** [comment]
Hi @arnouthpessoa, 

I'm really sorry for the inconveniences you faced.

TLDR: I have just discovered the reason. Remove the emoji from your filename, like '🏠' from ' 🏠 My Home'  and the program will run smoothly. The detail is mentioned below.

Bonus Tip: You can make a copy of the entire folder to a different directory (say in Desktop) and perform the operation ( removing emoji ), and then run the app from the new location. In this way, your main obsidian vworkflow ( filename with emoji ) will not get hampered. 

---

Here're some of my observations and possible cause for the problem. Do pardon and correct me if I misjudge anything.

- The program is designed to convert the .md notes to .html and host them using the framework. Now, sometimes, it may catch some errors while converting those files. In that case, it simply ignores that file and moves to the next markdown file in the file hierarchy.

- Now, this error can happen for various reasons depending from system to system. One of the main reasons is as follows: pubsidian only supports unicode characters (utf-8) of English language. After watching the screenshot you have sent, I think your .md files contain such character, which cannot be properly encoded using utf-8. Thus giving those errors. Can you please confirm, the difference between the notes, that have been converted and the ones that haven't. Do they carry different characters other than ASCII or Unicode (Like some emoji or other characters)?

- Another reason might be choosing the wrong attachment folder. Seeing your file structure, I think, this is your attachment (mainly images) folder for your obsidian vaults. Have you select this folder while prompted? Although this is the least possible reason of failure, but this can be happened.

Do let me know, if you still face any issue.

Amlan.

**Comment by arnouthpessoa** [comment]
Hey Mr @yoursamlan,

That's it! The Converter can only works with files that is in ASCII or Unicode. My whole vault have these emojis in filenames, and inside the files in queries, tags and etc. My PKM method uses these emojis. In order to use your Converter, I'll need to manually copy my files and create another structure without emojis interlinking them. Not a problem, knowing this from the start xD

I'll try later and come back to confirm to you.

Thanks again!

Arnouth.

**Comment by yoursamlan** [comment]
Dear @arnouthpessoa ,

That's really hurt to hear. I am adding the problem in the bugs list. Will let you know as soon as I fix the problem.

Sorry again for the inconvenience and thank you for pointing that out.

Amlan.
